module/zfs/zfs_ioctl.c

```diff
@@ -900,9 +900,18 @@ zfs_secpolicy_recv(zfs_cmd_t *zc, nvlist_t *innvl, cred_t *cr)
 	(void) innvl;
 	int error;
 
+	/*
+	 * zfs receive -F requires full receive permission,
+	 * otherwise receive:append permission is enough
+	 */
 	if ((error = zfs_secpolicy_write_perms(zc->zc_name,
-	    ZFS_DELEG_PERM_RECEIVE, cr)) != 0)
-		return (error);
+	    ZFS_DELEG_PERM_RECEIVE, cr)) != 0) {
+		if (zc->zc_guid)
```

Comment:
I think this would not handle `ZFS_IOC_RECV_NEW` scenario since `force` flag is embedded in `nvlist` there.

Reply:
Thanks for noticing, I just pushed a change to hopefully fix it.